plugins/scanners/scancode/src/test/kotlin/ScanCodeTest.kt

```diff
@@ -70,30 +65,13 @@ class ScanCodeTest : WordSpec({
         }
 
         "contain the values from the scanner configuration" {
-            val config = ScanCodeConfig.create(
-                mapOf(
-                    "commandLine" to "--command --line",
-                    "commandLineNonConfig" to "--commandLineNonConfig"
-                )
+            val scannerWithConfig = ScanCodeFactory.create(
+                commandLine = listOf("--command", "--line"),
+                commandLineNonConfig = listOf("--commandLineNonConfig")
             )
 
-            val scannerWithConfig = ScanCode("ScanCode", config, ScannerWrapperConfig.EMPTY)
-
-            scannerWithConfig.getCommandLineOptions("31.2.4").joinToString(" ") shouldBe
-                "--command --line --commandLineNonConfig"
-        }
-
-        "be handled correctly when containing multiple spaces" {
-            val config = ScanCodeConfig.create(
-                mapOf(
-                    "commandLine" to " --command  --line  ",
-                    "commandLineNonConfig" to "  -n -c "
-                )
-            )
-
-            val scannerWithConfig = ScanCode("ScanCode", config, ScannerWrapperConfig.EMPTY)
-
-            scannerWithConfig.getCommandLineOptions("31.2.4") shouldBe listOf("--command", "--line", "-n", "-c")
+            scannerWithConfig.getCommandLineOptions("31.2.4").joinToString(" ") shouldMatch
+                "--command --line --commandLineNonConfig --processes \\d+"
```

Comment:
Where's the "be handled correctly when containing multiple spaces" test gone?

Reply:
Parsing list options is now handled by the plugin API, I did not want to test its features in individual plugins. But I also realized that this also means that the arguments now have to be provided as a comma separated list, I'm not sure if that's good. I see three options:

* Kept it as is and require arugments as comma separated list.
* Make it a string option instead of a list option and do the splitting outside of the config class which is not so nice.
* Add a feature to the plugin API to allow defining a separator for list arguments, maybe as a regex like `\s+`.

Do you have a preference?

Reply:
> Kept it as is and require arugments as comma separated list.

I'm fine with this until we have a reason to change it. I just wanted the rationale to be recorded somewhere as part of the Git history.

Reply:
I have updated the `reference.yml` accordingly.